src/midi-player.ts

```diff
@@ -59,10 +59,29 @@ export class MidiPlayer implements IMidiPlayer {
 
             if (this._resolve === null) {
                 this._schedulerSubscription.unsubscribe();
+                this._schedulerSubscription = null;
             }
         });
     }
 
+    public stop(): void {
+        if (this._schedulerSubscription === null || this._endedTracks === null || this._offset === null) {
```

Comment:
It's a bit worrisome that the condition for stopped is not the exact negation of the condition of playing. To make it symmetrical, it may be best to create a method `_is_playing()` where the triple condition is evaluated, and used here and in `play()`.

Reply:
Thanks a lot for pointing this out. I've turned all the state-related properties into a single property to make the check easier to reason about and to prevent any invalid combination of the former properties.

https://github.com/chrisguttandin/midi-player/pull/358/commits/bea18e05c7f4c297114403fa02941bf8a0165524

Could you please take a look?

Reply:
Looks great, thanks!